File: DTAtoSEDAS/DatToSedasReengineer/LogMessage.vb
Imports System.IO
Imports DatToSedasReengineer.My
Imports Microsoft.VisualBasic.CompilerServices

Public Class LogMessage
    Private Shared _GlobalLog As Boolean = True

    Private Shared _GlobalOutputToConsole As Boolean = False

    Private Shared MessagePrefix As String = "Meldung"

    Private Shared WarningPrefix As String = "[Warnung]"

    Private Shared CriticalPrefix As String = "[**FEHLER**]"

    Private Shared _Path As String = Directory.GetCurrentDirectory() + "\DatToSedas.log"

    Public Enum MsgType
        Message
        Warning
        Critical
    End Enum

    Public Enum Output
        Console = 1
    End Enum

    Public Shared Property SetGlobalLog() As Boolean
        Get
            Return LogMessage._GlobalLog
        End Get
        Set(value As Boolean)
            LogMessage._GlobalLog = value
        End Set
    End Property

    Public Shared Property SetGlobalOutputToConsole() As Boolean
        Get
            Return LogMessage._GlobalOutputToConsole
        End Get
        Set(value As Boolean)
            LogMessage._GlobalOutputToConsole = value
        End Set
    End Property

    Public Shared Sub Show(msg As String)
        LogMessage.ShowMessage(msg, "", MsgType.Message, 1)
    End Sub

    Public Shared Sub Show(msg As String, type As MsgType)
        LogMessage.ShowMessage(msg, "", type, 0)
    End Sub

    Public Shared Sub Show(msg As String, output As Output)
        LogMessage.ShowMessage(msg, "", MsgType.Message, 1)
    End Sub

    Public Shared Sub Show(msg As String, type As MsgType, output As Output)
        LogMessage.ShowMessage(msg, "", type, 1)
    End Sub

    Public Shared Sub Show(msg As String, title As String)
        LogMessage.ShowMessage(msg, title, MsgType.Message, 1)
    End Sub

    Public Shared Sub Show(msg As String, title As String, type As MsgType)
        LogMessage.ShowMessage(msg, title, type, 1)
    End Sub

    Public Shared Sub Show(msg As String, title As String, output As Output)
        LogMessage.ShowMessage(msg, title, MsgType.Message, 1)
    End Sub

    Public Shared Sub Show(msg As String, title As String, type As MsgType, output As Output)
        LogMessage.ShowMessage(msg, title, type, 1)
    End Sub

    Public Shared Sub LogOnly(msg As String)
        LogMessage.WriteToLogfile(msg, MsgType.Message)
    End Sub

    Public Shared Sub LogOnly(msg As String, type As MsgType)
        LogMessage.LogOnly(msg, type)
    End Sub

    Private Shared Sub ShowMessage(msg As String, title As String, type As MsgType, output As Integer)
        Dim prefix As String = ""
        Select Case type
            Case MsgType.Message
                Dim flag As Boolean = Operators.CompareString(title, "", False) = 0
                If flag Then
                    title = "Nachricht"
                End If
                prefix = LogMessage.MessagePrefix
            Case MsgType.Warning
                Dim flag2 As Boolean = Operators.CompareString(title, "", False) = 0
                If flag2 Then
                    title = "Warnung"
                End If
                prefix = LogMessage.WarningPrefix
            Case MsgType.Critical
                Dim flag3 As Boolean = Operators.CompareString(title, "", False) = 0
                If flag3 Then
                    title = "Kritischer Fehler!"
                End If
                prefix = LogMessage.CriticalPrefix
        End Select
        Dim flag4 As Boolean = Not LogMessage._GlobalOutputToConsole And output = 0
        If Not flag4 Then
            Dim flag5 As Boolean = type = MsgType.Critical
            If flag5 Then
                LogMessage.WriteToConsole(prefix, msg)
                Console.Write("Bitte ENTER drücken...")
                Console.ReadLine()
            Else
                LogMessage.WriteToConsole(prefix, msg)
            End If
        End If
        Dim globalLog As Boolean = LogMessage._GlobalLog
        If globalLog Then
            LogMessage.WriteToLogfile(msg, type)
        End If
    End Sub

    Private Shared Sub WriteToConsole(prefix As String, msg As String)
        Dim array As String() = Strings.Split(msg, vbCrLf, -1, CompareMethod.Binary)
        Dim list As List(Of String) = New List(Of String)()
        Dim num As Integer = 80
        Dim array2 As String() = array
        ' The following expression was wrapped in a checked-statement
        For i As Integer = 0 To array2.Length - 1
            Dim text As String = array2(i)
            While Strings.Len(text) > num
                list.Add(Strings.Mid(text, 1, num))
                text = Strings.Mid(text, num + 1)
            End While
            list.Add(text)
        Next
        Dim num2 As Integer = list.Count - 1
        For j As Integer = 0 To num2
            Dim flag As Boolean = j = 0
            If flag Then
                Console.WriteLine(String.Format("{0,-12}: {1}", prefix, list(j)))
            Else
                Console.WriteLine("{0,-12}: {1}", "", list(j))
            End If
        Next
    End Sub

    Private Shared Sub WriteToLogfile(msg As String, type As MsgType)
        Dim array As String() = Strings.Split(msg, vbCrLf, -1, CompareMethod.Binary)
        Dim list As List(Of String) = New List(Of String)()
        Dim arg As String = ""
        Dim num As Integer = Strings.Len(DateAndTime.Now.ToString())
        Select Case type
            Case MsgType.Message
                arg = LogMessage.MessagePrefix
            Case MsgType.Warning
                arg = LogMessage.WarningPrefix
            Case MsgType.Critical
                arg = LogMessage.CriticalPrefix
        End Select
        msg = msg.Replace(vbCrLf, " ")
        Try
            File.AppendAllText(LogMessage._Path, String.Format("{0};{1,15} ;{2}", DateAndTime.Now.ToString(), arg, msg) + vbCrLf)
        Catch expr_AC As Exception
            ProjectData.SetProjectError(expr_AC)
            ProjectData.ClearProjectError()
        End Try
    End Sub

    Public Shared Function CheckLogFile(maxSizeKB As Integer) As Boolean
        Dim text As String = ""
        Dim list As List(Of String) = New List(Of String)()
        Dim text2 As String = ""
        Dim flag As Boolean = Not File.Exists(LogMessage._Path)
        Dim flag2 As Boolean
        If flag Then
            flag2 = flag2
        Else
            Dim num As Long = MyProject.Computer.FileSystem.GetFileInfo(LogMessage._Path).Length
            Dim num2 As Long = num
            Dim flag3 As Boolean = num2 > 1000L
            If flag3 Then
                num /= 1000L
                text2 = "KB"
            End If
            Dim flag4 As Boolean = num > CLng(maxSizeKB)
            ' The following expression was wrapped in a checked-statement
            If flag4 Then
                LogMessage.LogOnly(String.Concat(New String() {"Verkleinern der Logdatei von ", Conversions.ToString(num), text2, " auf ca. ", Conversions.ToString(num / 2L), text2}))
                Using streamReader As StreamReader = New StreamReader(LogMessage._Path)
                    text = streamReader.ReadToEnd()
                End Using
                Dim array As String() = Strings.Split(text, vbCrLf, -1, CompareMethod.Binary)
                Dim num3 As Integer = array.Count / 2
                Dim upperBound As Integer = array.GetUpperBound(0)
                For i As Integer = num3 To upperBound
                    list.Add(array(i))
                Next
                File.Delete(LogMessage._Path)
                text = ""
                LogMessage.LogOnly("######## Logdatei verkleinert ##########")
                Try
                    Dim enumerator As List(Of String).Enumerator = list.GetEnumerator()
                    While enumerator.MoveNext()
                        Dim current As String = enumerator.Current
                        text = text + current + vbCrLf
                    End While
                Finally
                    Dim enumerator As List(Of String).Enumerator
                    CType(enumerator, IDisposable).Dispose()
                End Try
                File.AppendAllText(LogMessage._Path, text)
            End If
        End If
        Return flag2
    End Function
End Class

File: DTAtoSEDAS/DatToSedas_CSharp/LogMessage.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DatToSedas_CSharp
{
    public enum MsgType
    {
        Message,
        Warning,
        Critical,
    }

    enum Output
    {
        Console = 1
    }

    class LogMessage
    {


        private static string MessagePrefix = "Meldung";
        private static string WarningPrefix = "[Warnung]";
        private static string CriticalPrefix = "[**FEHLER**]";
        private static string _Path = Directory.GetCurrentDirectory() + "\\DatToSedas.log";

        public static bool GlobalLog { get; set; } = true;
        public static bool GlobalOutputToConsole { get; set; } = false;


        public static void Show(string msg)
        { LogMessage.ShowMessage(msg, "", MsgType.Message, 1); }

        public static void Show(string msg, MsgType type)
        { LogMessage.ShowMessage(msg, "", type, 0); }

        public static void Show(string msg, Output output)
        { LogMessage.ShowMessage(msg, "", MsgType.Message, 1); }

        public static void Show(string msg, MsgType type, Output output)
        { LogMessage.ShowMessage(msg, "", type, 1); }

        public static void Show(string msg, string title)
        { LogMessage.ShowMessage(msg, title, MsgType.Message, 1); }

        public static void Show(string msg, string title, MsgType type)
        { LogMessage.ShowMessage(msg, title, type, 1); }

        public static void Show(string msg, string title, Output output)
        { LogMessage.ShowMessage(msg, title, MsgType.Message, 1); }

        public static void Show(string msg, string title, MsgType type, Output output)
        { LogMessage.ShowMessage(msg, title, type, 1); }

        public static void LogOnly(string msg)
        { LogMessage.WriteToLogfile(msg, MsgType.Message); }

        public static void LogOnly(string msg, MsgType type)
        { LogMessage.LogOnly(msg, type); }



        private static void ShowMessage(string msg, string title, MsgType type, int output)
        {
            string prefix = "";
            switch (type)
            {
                case MsgType.Message:
                    title = "Nachricht";
                    prefix = LogMessage.MessagePrefix;
                    break;
                case MsgType.Warning:
                    title = "Warnung";
                    prefix = LogMessage.WarningPrefix;
                    break;
                case MsgType.Critical:
                    title = "Kritischer Fehler!";
                    prefix = LogMessage.CriticalPrefix;
                    break;
            }


            bool flag4 = !LogMessage.GlobalOutputToConsole & output == 0;
            if (!flag4)
            {
                if (type == MsgType.Critical)
                {
                    LogMessage.WriteToConsole(prefix, msg);
                    Console.Write("Bitte ENTER drücken...");
                    Console.ReadLine();
                }
                else
                {
                    LogMessage.WriteToConsole(prefix, msg);
                }
            }

            if (LogMessage.GlobalLog)
            {
                LogMessage.WriteToLogfile(msg, type);
            }
        }


        private static void WriteToConsole(string prefix, string msg)
        {
            string array = Strings.Split(msg, vbCrLf, -1, CompareMethod.Binary);
            List<string> list = new List<string>();
            int num = 80;
            string[] array2 = new string[] { };

            // The following expression was wrapped in a checked-statement
            for (int i = 0; i < array2.Length; i++)
            {
                string text = array2[i];
                while (text.Length > num)
                {
                    list.Add(text.Substring(1, num));
                    text = text.Substring(num + 1);
                }
                list.Add(text);
            }

            int num2 = list.Count - 1;
            for (int j = 0; j <= num2; j++)
            {
                if (j == 0)
                {
                    Console.WriteLine(String.Format("{0,-12}: {1}", prefix, list(j)));
                }
                else
                {
                    Console.WriteLine("{0,-12}: {1}", "", list(j));
                }
            }
        }

        private static void WriteToLogfile(string msg, MsgType type)
        {
            string[] array = Strings.Split(msg, vbCrLf, -1, CompareMethod.Binary);
            List<string> list = new List<string>();
            string arg = "";
            int num = Strings.Len(DateAndTime.Now.ToString());

            switch (type)
            {
                case MsgType.Message:
                    arg = LogMessage.MessagePrefix;
                    break;
                case MsgType.Warning:
                    arg = LogMessage.WarningPrefix;
                    break;
                case MsgType.Critical:
                    arg = LogMessage.CriticalPrefix;
                    break;
            }

            msg = msg.Replace("\n\r", " ");
            try
            {
                File.AppendAllText(LogMessage._Path, String.Format("{0};{1,15} ;{2}", DateAndTime.Now.ToString(), arg, msg) + vbCrLf);
            }
            catch (Exception expr_AC)
            {
                Console.WriteLine(expr_AC.ToString());
            }
        }

        public bool CheckLogFile(int maxSizeKB)
        {
            string text = "";
            List<string> list = new List<string>();
            string text2 = "";
            bool flag = Not File.Exists(LogMessage._Path);
            bool flag2;
            if (flag)
            {
                flag2 = flag2;
            }
            else
            {
                long num = MyProject.Computer.FileSystem.GetFileInfo(LogMessage._Path).Length;
                long num2 = num;
                bool flag3 = num2 > 1000L;
                if (flag3)
                {
                    num /= 1000L;
                    text2 = "KB";
                }

                bool flag4 = num > CLng(maxSizeKB);

                ' The following expression was wrapped in a checked-statement
                if (flag4)
                {
                    LogMessage.LogOnly("Verkleinern der Logdatei von ", num.ToString(), text2, " auf ca. ", (num / 2L).ToString() + text2);
                    using (StreamReader sr = new StreamReader(LogMessage._Path))
                    {
                        text = sr.ReadToEnd();
                    }

                    string[] array = Strings.Split(text, vbCrLf, -1, CompareMethod.Binary);
                    int num3 = array.Count() / 2;
                    int upperBound = array.GetUpperBound(0);
                    for (int i = num3; i < upperBound; i++)
                    {
                        list.Add(array(i));
                    }

                    File.Delete(LogMessage._Path);
                    text = "";
                    LogMessage.LogOnly("######## Logdatei verkleinert ##########");

                    try
                    {
                        List<string> enumerator = new List<string>().Enumerator;
                        enumerator = list.GetEnumerator();

                        while (enumerator.MoveNext())
                        {
                            string current = enumerator.Current;
                            text = text + current + "\n\r";
                        }
                    }
                    catch (Exception ex)
                    { }
                    finally
                    {
                        Dim enumerator As List(Of String).Enumerator
                        CType(enumerator, IDisposable).Dispose()
                    }
                    File.AppendAllText(LogMessage._Path, text)
                }
            }
            return flag2;
        }
    }
}
